pulp_smash/cli.py

```diff
@@ -816,3 +819,143 @@ def _yum_apply_erratum(self, erratum):
     def apply_erratum(self, erratum):
         """Dispatch to proper _{self.name}_apply_erratum."""
         return getattr(self, '_{0}_apply_erratum'.format(self.name))(erratum)
+
+
+class RegistryClient:
+    """A container registry client on test runner machine.
+
+    Each instance of this class represents the registry client on a host. An
+    example may help to clarify this idea:
+
+    >>> from pulp_smash import cli, config
+    >>> registry = cli.RegistryClient(config.get_config())
+    >>> image = registry.pull('image_name')
+
+    In the example above, the ``registry`` object represents the client
+    on the host where pulp-smash is running the test cases.
+
+    Upon instantiation, a :class:`RegistryClient` object talks to its target
+    host and uses simple heuristics to determine which registry client is used.
+
+    Upon instantiation, this object determines whether it is running as root.
+    If not root, all commands are prefixed with "sudo".
+    Please ensure that Pulp Smash can either execute commands as root
+    or can successfully execute ``sudo`` on the local host.
+
+    NOTE: When running against a non-https registry the client configuration
+    `insecure-registries` must be enabled.
+    For docker it is located in `/etc/docker/daemon.json` and content is::
+
+        {"insecure-registries": ["pulp_host:8080"]}
+
+    For podman it is located in `etc/containers/registries.conf` with::
+
+        [registries.insecure]
+        registries = ['pulp_host:8080']
+
+    :param pulp_smash.config.PulpSmashConfig cfg: Information about the target
+        host.
+    :param tuple raise_if_unsupported: a tuple of Exception and optional
+        string message to force raise_if_unsupported on initialization::
+
+          rc = RegistryClient(cfg, (unittest.SkipTest, 'Test requires podman'))
+          # will raise and skip if unsupported package manager
+
+        The optional is calling `pm.raise_if_unsupported` explicitly.
+    """
+
+    def __init__(self, cfg, raise_if_unsupported=None, local=True):
```

Comment:
I am slightly confused about the use of ``local=True`` . Why can we just read from the ``settings.json``?

Or why can we add a ``pulp_host`` variable to the RegistryClient?
```suggestion
    def __init__(self, cfg, raise_if_unsupported=None, pulp_host=None):
```


I will play a bit more with this one. Perhaps add more notes for the docstring.

Reply:
The R`egistryClient` doesn't run on the machine where Pulp is installed. It runs locally on the machine where tests are executed.

If local = False, then it will read the host from settings and run on Pulp host.

I think it is best to default to `local` and do not have to pass the pulp_host on every run.